### PR TITLE
13522 - Global Key Command performance improvements

### PR DESCRIPTION
**Pull request by Cattlesquat**

### CODE REVIEW

**Review by uckelman — APPROVED** [review]

**Review by uckelman — APPROVED** [review]

**Comment by uckelman** [comment]
^^^ Conflicty McConflictface

**Comment by uckelman** [comment]
@BrentEaston Is there anything you want to look over on this one?

File: vassal-app/src/main/java/VASSAL/configure/BooleanConfigurer.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public void setName(String s) {
       box = new JCheckBox();
       box.setSelected(booleanValue());
       box.addItemListener(e -> setValue(box.isSelected()));
-      p.add(box);
+      p.add(box, "growx 0");
     }
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/configure/BooleanConfigurer.java:120]: Might be better to revert this change as I have already made this change in master, but done it differently.

File: vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java [hunk 2]
```diff
@@ -83,51 +101,303 @@ public void setReportSingle(boolean reportSingle) {
...
+              // This will iterate through actual game pieces
+              for (GamePiece gamePiece : pieceList) {
+                // If a property-based Fast Match is specified, we eliminate non-matchers of that first.
+                if (!passesPropertyFastMatch(gamePiece)) {
+                  continue;
+                }
+
+                // These basic location filters are faster than equivalent filters in the Beanshell expression
+                if ((target.targetType == GlobalCommandTarget.Target.ZONE) || (target.targetType == GlobalCommandTarget.Target.LOCATION) ||
+                  (target.targetType == GlobalCommandTarget.Target.CURZONE) || (target.targetType == GlobalCommandTarget.Target.CURLOC)) {
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java:313]: I don't think this if statement is needed. All options are covered in the immediately following switch statement. Just add a default: to the switch that does nothing.

File: vassal-app/src/main/java/VASSAL/build/module/map/DeckGlobalKeyCommand.java [hunk 3]
```diff
@@ -137,13 +150,18 @@ public void decode(String s) {
...
   }
 
   @Override
   public String[] getAttributeDescriptions() {
     return new String[]{
       Resources.getString(Resources.NAME_LABEL),
       Resources.getString("Editor.GlobalKeyCommand.command"), //$NON-NLS-1$
+
+      Resources.getString("Editor.GlobalKeyCommand.pre_select"), // Fast Match parameters
+
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/build/module/map/DeckGlobalKeyCommand.java:164]: I though we where removing Fast Match from DeckGlobalKeyCommand? If we do, the 'Additional Match Properties' label will need to revert to 'Matching properties'.

**Review by BrentEaston — CHANGES_REQUESTED** [review]

**Comment by BrentEaston** [comment]
Otherwise looks great. The Fast Match algorithm looks as good as it can be. Well done.

File: vassal-app/src/main/java/VASSAL/configure/BooleanConfigurer.java (hunk 1 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/configure/BooleanConfigurer.java:120]: reverted

File: vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java (hunk 2 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java:313]: Oh yeah, I think there was some earlier version of this with an intermediate step that made this make sense. Removed.

File: vassal-app/src/main/java/VASSAL/build/module/map/DeckGlobalKeyCommand.java (hunk 3 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/DeckGlobalKeyCommand.java:164]: good point

**Comment by Cattlesquat** [comment]
All cleaned up and perhaps ready-to-go
